src/pages/studyView/addChartButton/AddChartButton.tsx

```diff
@@ -303,6 +315,83 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
         }
     }
 
+    @computed get existSharedCustomData() {
+        if (this.props.store.isLoggedIn) {
+            return false;
+        }
+        return _.some(
+            this.props.store.customChartSet.entries(),
+            ([key, value]) => value.isSharedChart
+        );
+    }
+
+    @computed get notificationMessages() {
+        let notificationMessages: JSX.Element[] = [];
+        if (this.props.store.customChartSet.size > 0) {
+            // Notify if there any shared groups
+            if (this.existSharedCustomData) {
+                notificationMessages.push(
+                    <>
+                        <span
+                            className={classnames({
+                                [styles.sharedChart]: true,
+                            })}
+                        >
+                            Highlighted rows
+                        </span>{' '}
+                        are shared custom data.
+                    </>
+                );
+            }
+            // Notify that shared and page-session groups are not saved for non-logged users
+            if (
+                !this.props.store.isLoggedIn &&
+                AppConfig.serverConfig.authenticationMethod &&
+                AppConfig.serverConfig.authenticationMethod.includes(
```

Comment:
I think this makes 5 different places in the app where we make this check. Can we make a utility for this?